src/test/java/org/gridsuite/securityanalysis/server/SecurityAnalysisControllerTest.java

```diff
@@ -337,6 +341,51 @@ public void runAndSaveTest() throws Exception {
         assertResultNotFound(RESULT_UUID);
     }
 
+    @Test
+    public void testDeterministicResults() throws Exception {
+        MvcResult mvcResult;
+        String resultAsString;
+
+        SQLStatementCountValidator.reset();
+        mvcResult = mockMvc.perform(post("/" + VERSION + "/networks/" + NETWORK_UUID + "/run-and-save?reportType=SecurityAnalysis&contingencyListName=" + CONTINGENCY_LIST_NAME
+                        + "&receiver=me&variantId=" + VARIANT_2_ID + "&provider=OpenLoadFlow")
+                        .header(HEADER_USER_ID, "testUserId"))
+                .andExpectAll(
+                        status().isOk(),
+                        content().contentType(MediaType.APPLICATION_JSON)
+                ).andReturn();
+
+        assertRequestsCount(2, 6, 3, 0);
+
+        resultAsString = mvcResult.getResponse().getContentAsString();
+        UUID resultUuid = mapper.readValue(resultAsString, UUID.class);
+        assertEquals(RESULT_UUID, resultUuid);
+
+        Message<byte[]> resultMessage = output.receive(TIMEOUT, "sa.result");
+        assertEquals(RESULT_UUID.toString(), resultMessage.getHeaders().get("resultUuid"));
+        assertEquals("me", resultMessage.getHeaders().get("receiver"));
+
+        mockMvc.perform(get("/" + VERSION + "/results/" + RESULT_UUID + "/n-result"))
+                .andExpectAll(
+                        status().isOk(),
+                        content().contentType(MediaType.APPLICATION_JSON));
+
+        assertFiltredResultN();
+
+        var res = mockMvc.perform(get("/" + VERSION + "/results/" + RESULT_UUID + "/nmk-constraints-result/paged"))
+                .andExpectAll(
+                        status().isOk(),
+                        content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse().getContentAsString();
+
+        JsonNode resultsPageNode0 = mapper.readTree(res);
+        ObjectReader faultResultsReader = mapper.readerFor(new TypeReference<List<SubjectLimitViolationResultDTO>>() { });
+        List<SubjectLimitViolationResultDTO> subjectLimitViolationResultDTOS = faultResultsReader.readValue(resultsPageNode0.get("content"));
+        List<String> result = subjectLimitViolationResultDTOS.stream().map(SubjectLimitViolationResultDTO::getSubjectId).toList();
+        List<String> expectedResultInOrder = subjectLimitViolationRepository.findAll().stream().sorted(Comparator.comparing(o -> o.getId().toString())).map(SubjectLimitViolationEntity::getSubjectId).toList();
+        assertEquals(expectedResultInOrder, result);
```

Comment:
I think you can just test that the result and the result sorted are equals, no need to query again the DB

Reply:
It's not the same, I want to make sure that the result returned by the api call and the result retrieved and sorted directly by the DB is the same (for example if a row is missing we wouldn't know).

Reply:
alright :)